dependency/dependency_test.go

```diff
@@ -43,6 +43,20 @@ func TestMain(m *testing.M) {
 	tb := &test.TestingTB{}
 	runTestConsul(tb)
 	clients := NewClientSet()
+
+	defer func() {
```

Comment:
moved the defer here so we actually catch panics during setup, this won't catch panics during test runs, I spent some time trying to get to a working solution but it would require checking for running instances of consul/nomad/vault that did not previously get cleaned up before running